OptimizelySDK/Event/Builder/EventBuilder.cs

```diff
@@ -45,9 +46,12 @@ public class EventBuilder
 
         private Bucketer Bucketer;
 
-        public EventBuilder(Bucketer bucketer)
+        private ILogger Logger;
+
+        public EventBuilder(Bucketer bucketer, ILogger logger)
```

Comment:
make it nullable, the reason is if it is dynamically being used, then test cases will raise exception. 
and here 
logger =  logger ?? new DefaultLogger() assign.